tests/unit/test_sqlalchemy_utils.py

```diff
@@ -3,11 +3,13 @@
 import pytest
 import sqlalchemy
 from sqlalchemy.dialects.sqlite.pysqlite import SQLiteDialect_pysqlite
+from sqlalchemy.engine import Engine
 
 from databuilder.sqlalchemy_utils import (
     ExecuteWithRetry,
     GeneratedTable,
     OperationalError,
+    ReconnectableConnection,
```

Comment:
The comments within the new tests really helped me understand `ReconnectableConnection`.